### DISCUSSION TITLE
How to reconcile the videos in the library directory to the ones in the web gallery

### DISCUSSION THREAD

**Discussion by mystarkfuture** [immich-app/immich#3479] [Q&A]
Hello, 

I used the `immich-cli` app to upload a bunch of images and videos to the server that I have running in a cloud server. I have had no issue with the pictures - they are all there. 

`immich-cli` uploaded 680 videos. But the web gallery server stats page only shows 413 videos available. 

I've checked the library directory for the admin user - 

```
find . -type f | sed -e 's/.*\.//' | sort | uniq -c | sort -n | grep -Ei '(mov|MOV|mp4|MP4|mkv)$'
      3 mkv
      7 MP4
     45 MOV
    233 mp4
    390 mov
```
It shows a total of 678 videos available in the immich server data volume as shown above. 

I only have three file formats - MOV, MP4 and MKV as shown in the list above. I believe they are all supported in Immich. Besides, I don't see problem with any one format - i've randomly sampled files of each format and found that all of the random samples are there in the webapp (search using `m:<file name>`). Could this perhaps be a bug in the stat counter?

Is there are way for me to get the exact list of the files that are counted towards the server stat videos count shown below?
![image](https://github.com/immich-app/immich/assets/123919887/dd9122f7-b0d0-41a5-a249-c42cabb20461)

Things I've tested so far - 
1. I've checked for issues with reverse proxy and there are none - all of the files are there in the library directory in the data volume. 
2. I've tried restarting the transcoder job manually for all the files - it starts with the correct count - 680 videos, but it skips over huge chunks of files (already transcoded and the ones not showing up in the gallery) and completes the transcode quickly arriving at the same 413 videos figure. 

Would really appreciate any pointers towards identifying exactly which files are missing from the gallery. Then perhaps I've be able to maybe manually transcode them before upload or something. 

Any help is much appreciated.

**Comment by alextran1502** [comment]
Hello, when you trigger the Cli again, does it show the correct mismatch number of files?

**Reply by mystarkfuture** [reply]
Trigger CLI with all of the files still there like this?

`podman run -it --rm -v "/media/user/Files/media/all:/import" ghcr.io/immich-app/immich-cli:latest upload --key aaaaaaaaaaaaaaaaaaaaa --server https://immich.domain.com`

That is triggering all of the imports - all of the 4994 files are uploading. Perhaps when the job completes, i'll get the status? 

I'll update the status here after the job completes.

**Reply by coltenkrauter** [reply]
Any updates here? I am facing similar.
